src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
+
+    return response;
+  },
+};
```

Comment:
PrimathonAICodeReviewer

The `create` function in `APIResponse` should handle the case where both `data` and `error` are not provided. It should set `data` to `null` instead of an empty object.
```diff
-       response.data = {};
+       response.data = null;
```


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->